private/src/gst/gstavbvideosrc.cpp

```diff
@@ -97,6 +113,11 @@ gst_avbvideosrc_class_init(GstAvbVideoSrcClass * klass)
             g_param_spec_string("stream-name", "Stream name",
                 "Name of the stream", DEFAULT_STREAM_NAME,
                 (GParamFlags)(G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS)));
+
+    g_object_class_install_property(gobject_class, PROP_STREAM_TYPE,
+            g_param_spec_uint("stream-type", "Stream type",
```

Comment:
I still think that you can go the next step, and install a `g_param_spec_enum` here. So things look really cool on gst-inspect. Try for instance `gst-inspect-1.0 videotestsrc` and look at the `pattern` property. It really looks nice and simple to use. I know it's a bit more work, but I think is worth it. What do you think?

Reply:
Agreed.